lib/common/widgets/link_tile.dart

```diff
@@ -31,10 +34,13 @@ class LinkTile extends StatelessWidget {
         DiscoverType.User => Routes.user(id, info),
         DiscoverType.Review => Routes.review(id, info),
       }),
-      onLongPress: () {
-        if (discoverType == DiscoverType.Anime ||
-            discoverType == DiscoverType.Manga) {
-          showSheet(context, EditView((id: id, setComplete: false)));
+      onLongPress: () async {
```

Comment:
I try to use the new Dart switch expression whenever I can. I believe it would work here?

Reply:
I tried the following, but it didn't seem to work:
```dart
switch (discoverType) {
    DiscoverType.Anime || DiscoverType.Manga => showSheet(context, EditView((id: id, setComplete: false))),
    DiscoverType.User => await Clipboard.setData(ClipboardData(text: "$additionalData")),
     _ => {},
}
```

```
lib/common/widgets/link_tile.dart:39:11: Error: Expected to find 'case'.
          DiscoverType.Anime || DiscoverType.Manga => showSheet(context, EditView((id: id, setComplete: false))),
          ^^^^^^^^^^^^
lib/common/widgets/link_tile.dart:38:17: Error: The type 'DiscoverType' is not exhaustively matched by the switch cases since it doesn't match 'DiscoverType.Anime'.
 - 'DiscoverType' is from 'package:otraku/modules/discover/discover_models.dart' ('lib/modules/discover/discover_models.dart').
Try adding a default case or cases that match 'DiscoverType.Anime'.
        switch (discoverType) {
                ^
```

So I just opted for the good old switch case

Reply:
Aww man -_-